### ISSUE TITLE
Large blocks (>500 lumis/file) failing to be inserted in DBSWriter Go-based

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#11170]
**Impact of the bug**
WMAgent

**Describe the bug**
It's not really a WMCore issue, but it requires some effort from our side to identify the problematic blocks - maybe workflows as well - and follow up with whatever decision we come up with (like bypassing blocks/lfns in the agent dbsbuffer tables).

**How to reproduce it**
Run a workflow that creates more than 500 lumis in the same output file.

**Expected behavior**
This is a placeholder and the course of actions is still to be discussed.
However, to start with, after a chat with Valentin, we decided to start with:
1. run a test workflow in testbed and make sure that large output files are properly inserted into the database
2. scan all the agent log files (DBS3Upload) and list blocks that are constantly failing to be inserted.

**Additional context and error message**
It could be that the problem is much bigger and some of those blocks are actually accepted by DBSWriter, but run/lumi information is not available in the database. 

Here is a case of study, for a block that has been successfully inserted 2 days ago (from FNAL submit6):
```
2022-05-29 01:23:01,246:139988738467584:INFO:DBSUploadPoller:About to call insert block for: /DisplacedJet/Run2018D-UL2018_MiniAODv1_NanoAODv2_GT36-v1/NANOAOD#4756d791-0db1-4069-ba83-3f54ccd9bdd6
```
where I see the block and one LFN in the DBS database:
[https://cmsweb.cern.ch/dbs/prod/global/DBSReader/files?block_name=/DisplacedJet/Run2[…]v1/NANOAOD%234756d791-0db1-4069-ba83-3f54ccd9bdd6&detail=true](https://cmsweb.cern.ch/dbs/prod/global/DBSReader/files?block_name=/DisplacedJet/Run2018D-UL2018_MiniAODv1_NanoAODv2_GT36-v1/NANOAOD%234756d791-0db1-4069-ba83-3f54ccd9bdd6&detail=true)

but NO run/lumi data associated to it:
[https://cmsweb.cern.ch/dbs/prod/global/DBSReader/filelumis?block_name=/DisplacedJet/[…]oAODv2_GT36-v1/NANOAOD%234756d791-0db1-4069-ba83-3f54ccd9bdd6](https://cmsweb.cern.ch/dbs/prod/global/DBSReader/filelumis?block_name=/DisplacedJet/Run2018D-UL2018_MiniAODv1_NanoAODv2_GT36-v1/NANOAOD%234756d791-0db1-4069-ba83-3f54ccd9bdd6)
Unless I did something wrong, I think we indeed can end up with no run/lumi  even if the whole blockInsert is successful.

**Comment by amaltaro** [comment]
Quick update. I injected the following 2 test workflows in testbed:
amaltaro_TC_LHE_PFN_May2022_Val_220531_215238_6124
amaltaro_TC_LHE_PFN_May2022_Val_220531_220952_4891

and on what concerns blocks constantly failing to be injected into DBSServer, I found only 3 blocks for all the 4 agents (submit6, submit7, vocms0282 and vocms0283 - relval agent is fine!):
```
/BuToDStarX_ToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/RunIISummer20UL18NanoAODv9-Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM#b55338fc-46d1-4e97-b65f-1d09a9769c1f
/PMSSM_set_2_LL_2_TuneCP2_13TeV-pythia8/RunIIAutumn18MiniAOD-PUFall18Fast_GridpackScan_102X_upgrade2018_realistic_v15-v1/MINIAODSIM#ab3c8b71-80a3-461f-872c-43b2c797c3cd
/TTToSemiLeptonic_widthx0p85_TuneCP5_13TeV-powheg-pythia8/RunIISummer20UL16NanoAODv9-106X_mcRun2_asymptotic_v17-v1/NANOAODSIM#5954bf44-1a3d-4a9a-a589-6f502cf44a9f
```

**Comment by vkuznet** [comment]
@amaltaro do you have any update on your submitted workflows?

**Comment by vkuznet** [comment]
I'm glad that the first one passed the test. Regarding your question about sufficient testing. if you had 800 lumis, it will be split into two chunk, one chunk of 500 and another 300, and they will be executed concurrently. Therefore, it is sufficient for testing.

**Comment by haozturk** [comment]
@amaltaro asked me to provide some workflows and datasets which most probably were affected by this issue:
1. https://cmsweb.cern.ch/reqmgr2/fetch?rid=pdmvserv_task_BPH-RunIISummer20UL18NanoAODv9-00186__v1_T_220514_132751_5487
```
/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/RunIISummer20UL18NanoAODv9-Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM
```
2. https://cmsweb.cern.ch/reqmgr2/fetch?rid=pdmvserv_task_SMP-RunIISummer20UL16NanoAODv9-00168__v1_T_220516_014308_4277
```
/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/RunIISummer20UL16NanoAODv9-106X_mcRun2_asymptotic_v17-v1/NANOAODSIM
```
3. https://cmsweb.cern.ch/reqmgr2/fetch?rid=pdmvserv_task_SMP-RunIISummer20UL16NanoAODv9-00169__v1_T_220516_014310_3755
```
/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/RunIISummer20UL16NanoAODv9-106X_mcRun2_asymptotic_v17-v1/NANOAODSIM
```
Note that the impact is about 110 workflows.

**Comment by vkuznet** [comment]
@haozturk , it is very easy to check their status and all three datasets you listed have run-lumis in them, see:
```
cat datasets.txt | awk '{print "echo "$1"; ./dasgoclient -query=\"file,run,lumi dataset="$1"\""}' | /b
in/sh
/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/RunIISummer20UL18NanoAODv9-Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM
/store/mc/RunIISummer20UL18NanoAODv9/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/NANOAODSIM/Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/70000/1D2542F0-2B4A-B447-A517-FDAAFD421A33.root 1 [2454,2459,2464,2476,2477,2486,2492,2496,2499,2501,2508,2471,2472,2488,2489,2490,2498,2500,2504,2482,2483,2487,2502,2521,2522,2456,2458,2462,2474,2485,2511,2513,2514,2515,2463,2465,2466,2469,2475,2481,2495,2512,2516,2518,2461,2470,2473,2480,2491,2493,2494,2503,2453,2460,2479,2484,2497,2517,2455,2457,2478,2507,2519,2520]
/store/mc/RunIISummer20UL18NanoAODv9/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/NANOAODSIM/Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/70000/18455D41-6C7D-AB49-8DC0-1C139E0EED3E.root 1 [20792,20851,20783,20791,20808,20810,20823,20836,20838,20839,20843,20845,20829,20830,20832,20833,20834,20835,20846,20787,20790,20795,20799,20800,20803,20805,20815,20827,20848,20785,20804,20819,20820,20824,20825,20849,20853,20854,20789,20796,20797,20807,20816,20828,20837,20842,20788,20802,20809,20811,20831,20840,20844,20847,20850,20784,20786,20798,20801,20806,20812,20826,20841,20852]
/store/mc/RunIISummer20UL18NanoAODv9/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/NANOAODSIM/Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/70000/2AEC662D-AFE5-AA41-A1E4-BC3D10249843.root 1 [10006,10025,10065,10122,10128,10145,10173,10175,10176,10183,10219,10228,10302,10310,10365,10578,10588,10599,10611,10618,10640,10647,10648,10670,10693,11054,11186,11239,11240,11265,11266,11269,11292,11307,11343,11425,11470,11741,10008,10009,10017,10026,10035,10095,10118,10129,10250,10309,10373,10374,10432,10478,10479,10552,10579,10587,10617,10634,10639,10655,10694,10746,10769,10772,11030,11087,11088,11095,11268,11291,11815,10054,10080,10125,10199,10220,10232,10280,10284,10366,10477,10497,10521,10523,10524,10627,10743,11185,11201,11428,11446,11469,11509,11581,11742,11816,10010,10053,10146,10174,10301,10382,10423,10424,10427,10428,10498,10503,10560,10572,10609,10612,10623,10637,10657,10658,10717,10718,10744,11053,11181,11190,11193,11427,11580,11724,11814,11929,10018,10036,10096,10127,10149,10184,10191,10192,10231,10297,10367,10368,10369,10480,10504,10522,10580,10613,10624,10646,10669,11096,11119,11189,11202,11271,11272,11324,11344,11582,11707,11708,10005,10007,10058,10117,10143,10200,10214,10341,10342,10343,10409,10559,10577,10583,10597,10598,10600,10606,10638,10644,10651,10652,10653,10654,10770,10778,11029,11182,11308,11345,11445,11547,11548,10077,10079,10126,10130,10144,10150,10227,10258,10279,10283,10319,10320,10344,10410,10516,10551,10571,10584,10605,10610,10633,10641,10642,10656,10761,10762,10777,10779,11346,11352,11426,11510,11722,11723,11813,11925,11926,10057,10066,10078,10121,10213,10249,10257,10298,10370,10381,10431,10515,10614,10628,10643,10645,10745,10771,10780,11120,11194,11267,11323,11351,11721,11930]
/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/RunIISummer20UL16NanoAODv9-106X_mcRun2_asymptotic_v17-v1/NANOAODSIM
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/33E95F78-5C2D-6D4B-81B7-649C21953BE3.root 1 [11062,34511,125792,620,34493,34543,65854,125697,64762,125808,34407,34423,65666,162,17512,30663,34431,125459,6074,11244,19276,120713,34414,34527,125704]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/5D94A8A9-A6B9-EB44-A9C7-D7AB42D435C3.root 1 [3672,52484,52489,52602,52612,52613,52668,52671,52726,52751,52815,52816,52873,52921,62008,62019,62052,36140,52511,52551,52570,52667,52702,52750,52787,52806,52865,52889,3192,36777,52480,52493,52578,52621,52670,52744,52746,52758,52868,52956,36449,36549,36700,36827,52523,52527,52530,52539,52560,52561,52586,52687,52734,52926,3899,36222,36395,36548,36657,52481,52536,52594,52686,52759,52760,52841,52957,3260,3549,52488,52528,52620,52680,52725,52775,52825,52840,52980,3818,36343,36505,36719,36823,52526,52579,52603,52652,52724,52735,52742,52782,52833,52885,52897,52925,52979,62013,3148,36083,36230,36403,36511,36691,36744,52733,52741,52743,52839]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/8D3F86A4-D718-3E40-A6FC-E51E18EBC83F.root 1 [5614,5657,5663,5740,5744,5745,5763,18112,18215,18378,18403,21557,21567,25378,25415,25423,25547,25793,26276,31283,31385,31501,31515,31534,31837,43068,51023,51348,5294,5631,5831,5839,5928,18099,18147,18279,18311,18385,25411,25422,25618,25652,25807,31160,31227,31302,31308,31332,31571,31753,43022,51074,51142,51295,51356,65451,120221,120787,5623,5630,5677,5846,12613,14057,18097,18114,18226,18232,18479,21545,25220,25660,25739,26091,26150,26163,26329,31163,31458,31512,31586,31652,31803,5607,5626,5628,5634,5653,5774,5833,11813,11959,18107,18201,18409,18429,21531,21549,21558,25213,25601,26056,31138,31161,31443,31531,31554,43739,43752,43846,51362,65206,120790,5646,5658,5666,5668,5675,5682,5739,5751,5778,18106,18236,18248,18260,18282,18291,18316,18388,21532,25248,25716,26128,26335,31364,31539,31953,43741,51278,51291,64110,64116,5602,5609,5619,5635,5648,5676,5737,5741,5898,5989,11785,14015,14026,18101,18416,21525,25403,25559,25639,25681,25780,26195,26225,31044,31170,31320,31647,31789,31893,31924,31942,43743,5601,5738,5988,5998,14027,14054,18269,18341,18352,18476,25043,25903,26048,31100,43747,51314,120779,5088,5606,5625,5640,5660,5683,5731,5893,5900,5919,5920,18108,18149,18392,18394,21540,25061,25770,26013,31327,31366,31624,31760,64369]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/A9CF0C6D-956A-824C-B784-2992A1C2FAE5.root 1 [4625,5055,11091,11138,12351,12679,13953,18304,18423,24223,24637,24671,26524,31550,35144,35352,35633,35735,36029,60255,69071,69580,69940,75457,77153,77929,81894,89199,91687,91688,101571,110690,121202,121236,1056,2538,2846,2865,12539,12774,12776,12777,18307,18366,18783,23153,23548,23994,35650,35854,45999,46327,47237,53246,63616,64715,75563,75567,75569,77456,81630,81924,86498,89914,89915,89917,91498,92268,93821,94246,108866,121025,125583,1132,2863,7445,12160,12773,12778,18211,18306,18779,18793,23641,23993,24640,26559,35345,35354,35536,35871,45995,47097,47243,63623,63645,68012,69072,69128,69176,69944,75568,76091,77157,77858,85128,86495,89212,89921,91675,101570,110158,110829,121044,788,2807,7414,13922,18682,22025,26620,26846,35883,36273,36814,50057,50983,51011,53219,53350,53425,54295,54554,60232,60620,63653,63817,69057,69449,69552,69563,75208,75458,77120,77817,81906,81926,86255,89206,89210,94486,101569,101848,110594,110612,110824,116202,5057,5961,11117,12008,12174,13942,18784,22209,26520,26666,35533,35872,43899,46329,47101,51015,51028,53361,55102,69064,69083,69089,69460,69811,69813,77143,81896,81907,81923,82074,82076,82127,89216,95338,110513,116378,121204,121218,121322,2553,2620,2621,7436,7456,12155,18303,18373,26646,26672,37459,43898,45986,47822,53247,63429,69773,75441,75454,75570,76096,77560,82309,86318,86492,89516,101130,101851,108956,116079,116677,116748,121031,121206,2840,4647,5929,12771,18790,18791,24212,24222,26695,35532,36896,43909,45987,45991,46184,46193,46328,46342,53882,55605,60249,63650,75460,76347,81316,85814,91649,101568,101847,116054,116682,121053,121534,2861,4166,7449,12012,18305,18438,22178,26659,26847,32289,32827,45993,46338,47096,47110,51124,53248,53789,60959,63819,63909,65301,69815,69947,75564,81379,81904,81914,81920,84414,85794,86494,89204,89214,89912,89916,101845,110094,116204,121024,121039]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/A0BB4881-2496-FF48-AF2B-20958599E90A.root 1 [60346,60359,76158,86171,86758,91058,92548,93948,94292,116266,116330,60314,60444,60477,75242,86211,91024,101366,110445,116322,116527,116996,60120,75178,75484,75903,85678,86931,89261,89343,91598,91804,93924,94015,101481,101606,110770,60279,75647,85230,86312,89482,92764,93053,94093,101706,116300,121986,60280,60696,60752,86165,86358,86973,91600,92928,94099,94338,94433,94988,101364,101872,101984,116178,116964,116999,121995,60601,75032,75434,75498,86179,86212,86400,86571,89026,89041,91114,91166,92962,93425,116521,60297,60639,65255,75465,76603,91014,92974,93049,93213,94192,94330,110176,110237,120142,60282,60511,76790,89184,92609,93373,93429,93935,94096,116249,116364,116998]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/1053BC2C-DDF4-1E4C-A5D7-999BC9D510C3.root 1 [56864,59962,70975,71000,74618,74708,74837,79426,79647,83237,83502,83589,89757,105084,106913,107397,112310,124180,124590,124613,27846,35124,48155,48301,54072,54100,56825,58366,59758,59828,59997,70234,74723,79310,79391,79588,79649,79698,79793,83040,83526,96213,96301,96336,96369,96496,96722,96893,97593,102145,105154,106914,112795,118830,118924,124029,124063,126618,126916,127558,48275,70514,70675,76845,79207,79694,83775,94161,96239,97559,97612,102080,105016,105263,105265,105459,106915,118893,118927,118940,8270,47198,54560,56884,59730,59837,70688,97310,102027,105233,105600,106916,106981,118123,124483,124487,8196,10159,20770,27308,45525,46187,48249,48271,58393,58398,58768,58778,70108,70924,74470,74673,74726,74905,79535,79742,83621,102007,102116,105086,106119,106924,118733,118869,124663,124824,126269,126970,126994,45698,54547,59885,70023,74829,74855,79236,79587,79663,79736,79822,83053,96225,96984,97258,102146,105065,106904,106909,106919,118233,124488,124490,124757,126617,35844,35988,45227,45231,47079,47098,47244,48171,56887,70144,79237,79821,79990,83269,83801,83846,87281,96315,96937,97378,105071,105581,106105,106110,106903,106905,106906,106910,106942,106956,124997,126060,126790,48130,56997,59944,70553,74105,74598,79638,79696,79756,83757,91903,96049,96278,96344,96693,96730,96820,96921,102026,102147,105200,105466,105776,106911,106920,106935,106941,112740,126211]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/D05A6541-C1BB-834A-8E37-3A64DA55E06E.root 1 [31638,53330,77855,85219,93470,94240,108785,108859,115668,11751,51045,77186,79451,92898,101844,105928,106940,26054,53526,65509,65683,77867,86983,94041,9444,26660,43919,65311,77454,79450,94271,94676,120074,11596,11784,22882,26994,38817,64449,75668,77347,77725,81130,85476,121596,121766,11697,51792,94127,110514,113604,120611,121139,121588,22887,31724,60612,65013,65882,77132,86394,115324,116901,120220,121125,9544,22161,77291,77871,94485,101177,110052,110944,121243]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/A9740855-5F6C-4641-AEAA-CB0F16310BFD.root 1 [66287,62504,67783,42042]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/827AD7C8-33D1-A649-B177-66E9F69E2329.root 1 [8054,15132,15484,15648,25723,53643,58228,58381,58429,58434,58568,59013,59067,59101,60144,60241,60845,64262,70885,74771,74871,74928,76265,78217,78380,78655,91429,91621,91648,91650,93262,96271,102424,102585,102621,105258,105696,107680,107682,107684,107689,112767,112780,112871,113955,115199,124275,127023,127039,8132,8560,15717,31005,59076,60138,60476,70798,74869,78205,78596,78627,78678,79483,91051,91189,105489,106875,106884,112749,113729,113731,113909,115605,115725,115768,115958,118092,126697,126702,8115,9569,9982,15046,15472,20008,20175,58437,58668,59103,60607,78008,78432,78455,78615,78798,78820,91044,91430,96267,96272,105720,106831,106843,106844,106862,106893,106931,112899,113640,121519,124020,124073,124145,126137,126873,127054,8946,15770,25755,59146,59341,59375,60242,60848,70882,74495,74937,78032,78112,78332,78536,78872,78969,79557,79827,91053,91154,102493,105705,106847,106873,106890,107681,115356,115607,115719,118059,121524,121689,124252,126285,127041,127064,8059,10281,20182,26931,58227,58541,58684,59286,59348,59353,60132,60978,60987,78053,78155,78245,78386,78424,78925,79545,83999,96104,105375,105487,105529,105694,106818,106833,110681,113564,113953,115026,115463,115657,115715,115772,115937,124072,124245,126415,126703,126872,8096,8468,10106,15743,58226,59053,59258,65823,70698,74936,76102,76534,76883,78231,78443,78887,102495,102502,102531,105530,105695,106823,106828,106840,106854,106859,107679,112791,112817,115833,115941,120167,126050,126417,8320,8553,8561,8761,9103,22788,25655,25724,58223,58436,59284,60226,60420,60509,60840,70884,78152,78481,78592,78624,78663,78856,78862,83227,89203,91137,91713,96268,102497,102523,102622,105765,105813,106850,106861,106868,106888,107683,107686,113741,115926,126130,126132,126133,126416,126701,126878,126950,8060,8066,10485,15244,58435,59150,59163,59247,60461,60597,60715,74764,76087,78016,78142,78179,78191,78376,78402,78433,78438,78838,78911,79774,91831,102599,105565,105987,106834,106838,106865,106922,106932,107678,113418,115034,115428,115523,115572,115848,120984,124423,126129,126136,126870,126874,126875]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/0AE79E51-69CE-D440-9EB4-027E60144CCF.root 1 [5113,8625,9737,11268,11541,11682,18772,21149,22568,23977,23996,24706,26755,43311,43771,43973,50797,51145,56149,60914,64511,73028,73139,75748,77116,77968,79418,84180,85143,91919,92899,94061,94607,94782,97622,101330,101832,110897,113387,116191,120173,120413,120422,121108,121975,127174,5528,5638,8049,11632,11961,12172,12953,12963,15119,21161,22113,24157,24894,31173,31174,44468,45619,48083,50452,51779,53773,57963,64642,65381,65785,65927,76112,77562,79437,80556,81079,83349,84176,85149,85813,89104,92011,92633,96830,100997,101296,106912,107153,116825,120470,120514,120564,120566,120818,121162,121259,122951,126481,9881,9914,12242,15726,18698,20102,20796,21028,22277,22455,22677,23986,24064,24123,24528,24569,27029,33606,33674,43083,43871,45684,49826,51786,53370,54727,56351,56506,60355,75552,77208,77257,77574,77814,79233,80455,81059,81224,85433,86948,91848,100082,101118,106855,108699,113514,116445,116752,120461,121413,124884,7361,10259,11045,14842,15940,21029,21034,21671,23984,31558,33334,35582,37114,37690,48259,51104,53413,53423,57346,57364,59368,60365,60445,76215,76498,77078,77268,77452,81661,82969,83091,89091,89687,89777,92236,97602,97887,102857,105056,113620,116187,116545,120978,9228,15705,21027,21030,21159,22347,23275,26643,41097,43242,45294,47993,48743,48911,57857,59989,64104,64260,64281,68976,73149,75380,76414,81138,85962,85965,87996,91292,91414,93090,95524,101373,107283,110516,116086,120176,120521,121798,10503,11526,21143,22447,22532,22935,26833,33707,33869,37716,41461,43316,44223,53587,53782,56057,57186,64216,64221,65575,70111,77730,80888,83396,83714,89436,89536,93364,94284,101305,106173,115055,120697,124097,5309,11066,11501,11635,11858,22143,24477,24536,24944,25989,42716,43011,49795,51534,53987,56682,57714,57813,57950,59809,75452,76465,79767,80836,81128,83309,84242,85575,88471,92292,95909,101026,110703,113774,116663,120917,121306,121344,126405,126938,127458,18763,20227,20977,21807,23019,24236,25126,26563,27179,37713,43091,43807,51890,57749,57769,58255,60610,65896,75543,76032,77876,78839,80550,84635,86264,91905,92090,96728,96821,97875,100108,105731,107031,108274,110873,116525,116643,120908,123860,126421,127005]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/2D91A241-854F-4D4A-9367-A8DCFAE0E951.root 1 [68207,90160,90517,90705,95115,95469,95631,98027,98473,104092,109886,109991,111485,111517,111534,111542,111558,111583,68084,68366,68874,68918,90887,104195,104945,109313,111546,111575,111616,111624,61857,68230,68578,68821,88220,88281,90947,98090,98752,104422,104453,109894,111026,111091,111494,111663,68104,68459,68582,88863,98630,104447,104849,109158,109756,109911,109933,109975,111108,111476,111526,68534,68641,72040,88214,88312,90257,90403,98842,104729,109443,61979,68321,68478,68644,88133,88339,88517,90627,90681,95155,95197,95512,98097,109968,111647,88257,88381,88630,88637,95381,95702,104024,109866,109993,111149,111671,68058,68086,68506,68820,88593,88722,90542,95647,95715,104659,111665]
/store/mc/RunIISummer20UL16NanoAODv9/DYJetsToMuMu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/4AD63E5D-2DC7-5C45-8430-DAB923969725.root 1 [1242,2270,2271,2421,2783,14176,14703,21017,21148,21270,21857,21968,21995,23257,23264,23547,23868,26437,26506,26647,26819,31352,31582,31848,49560,50542,51088,51640,51876,53880,63081,63655,77160,86607,86796,92884,92890,101310,2525,2598,2820,14714,14718,21551,21966,23022,23177,23227,23255,23317,23496,23500,23505,26856,31280,43095,43672,49100,49104,51898,63124,63648,63850,63863,63911,77495,77515,77529,85833,86644,86982,101267,2427,2429,2434,2743,2808,2943,13442,13789,14704,14708,21754,23135,23272,23428,23755,23897,26976,43366,43461,43724,51780,63357,63470,81938,86267,86427,92887,101392,1739,2005,2433,2794,2843,13013,14273,21632,21849,23308,23748,23892,26421,49085,49440,49480,49549,50700,51293,53391,57949,63066,63373,63379,63882,65053,75491,75852,85222,85511,85704,85893,86089,86500,86624,86695,101360,1114,1426,2013,2017,2269,2930,13453,14747,21719,21825,21989,23027,23253,23465,26543,43566,43568,49455,50742,77155,77426,77692,85526,86655,101334,101396,1063,1104,1116,1514,1728,2422,2821,13769,14274,14643,14706,23126,23483,43397,49195,49852,63078,63089,63212,63386,63615,63656,63816,75258,75437,77362,92566,101307,101344,120355,1138,2006,2273,2460,2786,14175,14270,21032,21647,23320,23460,23983,26767,26859,31051,31224,31485,43013,43348,43902,49330,49446,50784,51492,57198,63203,63295,63305,63368,63455,63651,63818,65858,77436,85862,86002,86030,86263,86359,86611,86648,92805,92894,101332,120313,2010,2267,2798,2806,2839,13219,13220,13744,13975,14226,14739,21156,21157,21720,21786,21980,23277,25845,26460,26721,26975,31903,43096,49040,63006,63464,63950,63982,75050,75331,77305,77584,77834,85153,86924,86998,101370]
/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/RunIISummer20UL16NanoAODv9-106X_mcRun2_asymptotic_v17-v1/NANOAODSIM
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/8E2861DD-FD84-9848-9F77-DCFE9A4ED224.root 1 [55565,55686,59063,59908,62596,62688,70425,70748,73083,73612,55769,58787,62035,62197,62367,62627,62764,65019,65575,70437,79605,79797,79889,55807,55922,59983,62620,65540,95186,55434,55472,55639,70166,70551,95185,62046,62362,62605,62766,62769,65055,65326,65602,70106,55641,55785,55959,58258,59352,62359,70160,70309,70725,70758,73198,95122,95124,55231,55247,55712,55856,55980,58041,58138,58267,59322,62356,70002,70728,79848,95055,55563,55784,55880,58260,59917,59969,59984,62009,62363,65461,70380,70670,73074]
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/13D62031-95DB-AA43-8772-8402C6A08A8A.root 1 [7097,7120,7124,7699,9212,9288,9433,9472,12870,149471,149603,149788,149867,161001,161018,161033,161063,161074,161219,161221,166429,166983,170491,170628,247309,248024,265542,265600,265623,265941,271005,271055,271212,271736,271860,281139,7115,7125,7237,7517,7703,9151,12129,12311,14372,14990,44996,149503,149598,149620,149854,149876,161020,161022,161054,161067,201006,247110,247157,247238,247651,247652,247761,247980,248038,248302,248303,248517,260473,260544,260695,260842,265575,265651,265958,271213,271549,271556,271684,271864,279024,279138,279209,281046,281233,281321,281361,7158,7178,7217,7246,7557,7576,9461,9535,9582,12246,12329,149790,161038,166393,170630,201007,201018,201073,201080,201089,247141,247168,247197,247346,247448,247465,247598,248065,248384,260756,260878,260881,265338,271219,271467,271499,271523,281223,281253,281390,281485,7107,7118,7129,7461,7462,7495,7560,7574,7694,9188,9455,9610,12607,14589,14685,17477,149470,149607,149659,149787,149799,149853,166275,166787,166988,166996,170541,201049,247158,247188,247199,247750,248365,248527,248653,260546,265673,265858,265917,271415,271672,271727,281125,290923,7100,7108,7223,7505,7540,7731,9749,12072,149475,149635,149802,149822,161012,161019,161235,165596,166815,247176,247771,247804,247857,247917,248198,248273,248631,260586,260598,265601,265766,271266,279073,281350,281377,7063,7116,7117,7890,7936,9378,9445,9726,12306,14377,17309,17547,149021,149511,149616,149621,149791,149792,149808,149826,149839,149841,166384,166986,166997,247182,247291,260758,260814,265141,265235,265305,265480,265617,265642,265722,265754,265957,271204,271276,281030,281209,281410,281467,290691,290756,290910,7106,7409,7765,9513,12060,12723,14605,14978,14995,17486,149261,149625,149777,161044,166379,166431,166974,166999,170559,247143,247302,247348,247440,247636,247663,247725,248362,260684,260825,265280,265374,265761,271192,271373,271515,271717,271835,281316,7096,7099,7336,7390,7561,9618,14680,149501,166321,166394,167000,170506,170531,201055,247172,247173,247381,247470,247856,247966,248371,248606,260500,260618,260632,260637,260658,260660,260685,260732,265210,265321,265406,265502,265681,265973,271023,279014,281084,281263,281320,281349,290695,290709,290755,290948]
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/1232FD0F-D407-B548-BC93-D06BE5AD3CB2.root 1 [1074,1932,1278,26001,1339,26289]
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/3D7CA80B-C408-F048-A956-C3DFB0AFEB24.root 1 [175959,175963,175966,176033,176146,176783,176887,176891,181108,181905,182370,182380,182398,182426,182596,182687,205200,209639,210521,210593,211401,211406,211425,211427,211444,211446,211447,211448,224338,224384,224517,224529,224728,233588,233810,233915,238076,238125,175989,176773,176919,182388,182391,182425,182458,182607,182661,205272,209424,209563,210266,210311,210341,210366,210563,211028,224315,224563,224596,224616,224696,224846,224848,233304,233612,233806,233934,233960,238004,238059,238156,175961,175967,176174,176757,182374,209375,209501,209622,209627,209631,209636,211399,211433,211439,211440,224305,224637,225045,233614,233627,233938,233943,175949,176100,181912,182379,182429,182551,182589,182622,182636,205211,205362,209329,210546,210555,224306,224319,224344,224398,224497,224526,224607,224638,224700,233342,233878,238152,175948,175952,175955,175956,175958,176774,182554,182631,205360,205365,209336,209341,209394,209470,209585,209634,209638,209642,210589,211019,224349,224531,224575,224600,224617,224619,224733,233081,233527,233816,233871,238158,175951,175962,175978,175983,176001,176886,176902,182386,182428,182431,182436,182447,182598,182667,182669,209292,209411,209502,210285,210537,210541,210551,210675,211080,211420,224307,224323,224435,224498,224565,224626,224652,224688,233287,233290,233720,233912,175950,175968,176067,176147,176842,182382,182400,182613,205212,205218,209441,209567,209581,209623,210289,210421,210687,211424,224308,224372,224618,224713,233724,233782,233839,233850,233890,175947,175960,175965,176004,176104,176899,181909,182387,182392,182416,182605,182608,182679,182697,205154,205367,205369,205429,209447,209572,209586,209587,209628,209674,210288,211443,224309,224311,224322,224343,224704,233028,233269,233317,233328,233517,233996]
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/3E3DEB7D-E054-944E-83E0-469BCAE79188.root 1 [38014,44118,44787,1489,42532,85483,3648,42833,1143,121064,26075,26775,39255,44098]
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/EFD4AF35-3F61-304D-9B39-8F0252E4B186.root 1 [5519,10365,58968,59456,59621,59646,59693,59715,59826,62095,85310,108044,108240,110210,112656,112747,112894,121421,121474,121727,284365,284995,290355,5840,59313,59340,59364,59368,59585,59691,59756,62021,62089,77269,77309,108232,110406,110434,120425,120529,120598,251253,255174,2014,10687,58011,59502,59598,59721,59867,85550,102075,108959,114023,114308,114342,121588,290571,5543,11827,58038,58071,58171,58182,59443,59567,59667,59723,77334,85044,109000,110036,120959,121604,121659,169082,256387,262706,11982,58019,59608,59690,77273,85747,87203,114449,121299,121859,273909,290031,291009,11745,48319,49187,59338,59861,91059,102205,110513,112885,255164,267909,272197,284235,288807,290292,2657,5445,5455,5854,6112,58128,59341,59442,59453,59474,59663,59709,77121,77271,85236,87348,110035,112859,120378,256616,284377,5673,5675,32903,59579,59720,77290,77331,77674,82191,108069,110248,112703,114464,120967,121637,255531,285168,290524]
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/D114AEA4-4BCF-B744-A4D1-0664BE950289.root 1 [13226,34363,93630,96127,105053,105426,115829,147465,153452,153954,158794,173359,174035,174204,175364,175436,176901,176956,181231,181259,181422,181429,182157,182298,183265,243164,243199,19399,27479,38290,38325,38857,40095,83612,83711,93068,93779,94353,105074,115217,115690,141422,141672,141725,146970,147305,147559,150721,153011,153197,153445,153678,158416,158761,173134,174688,175618,176448,181428,182270,182593,183912,243209,27570,27643,29310,33529,40513,115116,140824,146101,146698,150978,152014,153681,158124,158862,173088,173206,174579,174674,174756,175199,175837,176073,176115,181705,181962,182474,188002,188706,189193,192565,33313,33690,40960,83969,84656,93246,96691,104415,106291,118009,118273,118488,146174,150682,153781,171552,173304,174725,174978,175466,176080,176811,181411,181984,183267,190350,190825,190980,236155,33121,33390,33551,37362,38538,84834,92865,105564,115226,127615,139736,140911,141581,146462,146601,146729,147260,147351,147725,150467,150671,152023,152049,153096,153848,158283,174749,174886,176937,181439,181943,182434,191630,236180,5895,6311,11505,28701,33339,33651,84273,92944,93491,94591,96455,147115,150332,150532,152292,153217,158429,159789,173237,173788,174159,174384,175113,175209,176460,181555,181953,181954,183339,186013,186101,186480,190916,192784,236192,243225,243569,13097,27025,33517,38766,94426,96569,104666,139149,147217,147758,150417,150696,150948,158081,158843,159137,171338,173856,174238,175833,176290,182557,186675,186726,191281,192300,192652,5437,27417,33592,33947,34535,94275,118085,127064,141864,141995,147254,147401,147731,147861,153515,155118,156476,158613,158991,171996,173517,175439,175648,176277,176367,176376,183176,189224,190166,191218,192318,192425]
/store/mc/RunIISummer20UL16NanoAODv9/WplusJetsToMuNu_H2ErratumFix_TuneCP5_13TeV-powhegMiNNLO-pythia8-photos/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/2830000/AE5B8DA9-1F1E-1744-99A2-A0F60CAB5233.root 1 [254180,256286,256623,256892,267827,267952,288273,288564,292070,254096,256121,256254,256294,256586,256695,256746,256757,256853,256861,267195,267341,267393,288244,288984,292264,292509,292523,254047,254098,256729,256933,267197,267942,288539,292253,254179,256356,256747,256886,267184,267427,267542,288547,288706,256235,256366,256671,256794,267157,267165,267673,288787,292480,256162,256479,256753,256785,256801,256850,267564,267656,267878,267976,288183,288395,288646,288743,292375,254040,254097,256735,256870,267602,267699,267935,292019,292461,256851,267962,288515,288731,288867,288906,292308]
```

**Comment by amaltaro** [comment]
Discussing with people here and there, the following actions are still to be taken:
1. Valentin will list all LFNs inserted into DBSServer given a creation time (>= May 17th). With that list, he will check whether they have any lumis in the DBS database. Once we know which files were incorrectly inserted, we need to map them to a dataset and then to a workflow and decide how to proceed (likely involving P&R/PdmV)
2. Alan/Todor need to mark the 3 blocks reported in this comment https://github.com/dmwm/WMCore/issues/11170#issuecomment-1142605703 as injected. Otherwise the agent will keep trying to inject them forever.
3. Hasan/DM team need to invalidate these 3 blocks above.
4. Due to the invalidation of the 3 blocks above, Hasan/PdmV need to evaluate whether a new workflow is needed.

Is there anything else missing?

**Comment by vkuznet** [comment]
Per my assignment I did the following:
- obtain timestamps for data look-up
```
python
>>> date_String = "17/May/2022:00:00:00 UTC +0000"
>>> dt_format = datetime.datetime.strptime(date_String, '%d/%b/%Y:%H:%M:%S %Z %z')
>>> dt_format
datetime.datetime(2022, 5, 17, 0, 0, tzinfo=datetime.timezone(datetime.timedelta(0), 'UTC'))
>>> time.mktime(dt_format.timetuple())
1652760000.0
>>> date_String = "01/Jun/2022:14:00:00 UTC +0000"
>>> time.mktime(dt_format.timetuple())
1652760000.0
>>> dt_format = datetime.datetime.strptime(date_String, '%d/%b/%Y:%H:%M:%S %Z %z')
>>> time.mktime(dt_format.timetuple())
1654106400.0
```
- query DBS database for data using the following statements (here I show all queries I did to get list of datasets/blocks/files with run-lumi counts and time-range numbers from above step):
```
SET LINESIZE 500
SET NEWPAGE 0
SET SPACE 0
SET PAGESIZE 0
SET ECHO OFF
SET FEEDBACK OFF
SET VERIFY OFF
SET HEADING OFF

SPOOL /tmp/d.txt
SELECT dataset
FROM cms_dbs3_prod_global_owner.DATASETS
WHERE creation_date>1652760000 AND creation_date<1654106400;

SPOOL /tmp/b.txt
SELECT block_name
FROM cms_dbs3_prod_global_owner.BLOCKS
WHERE creation_date>1652760000 AND creation_date<1654106400;

SPOOL /tmp/fl.txt
SELECT
COUNT(FL.RUN_NUM) AS NRUNS,
COUNT(FL.LUMI_SECTION_NUM) AS NRUNS,
F.LOGICAL_FILE_NAME AS LOGICAL_FILE_NAME
FROM cms_dbs3_prod_global_owner.FILE_LUMIS FL
JOIN cms_dbs3_prod_global_owner.FILES F ON F.FILE_ID = FL.FILE_ID
WHERE 
F.creation_date>1652760000 AND
F.creation_date<1654106400 AND
F.is_file_valid=1
GROUP BY F.LOGICAL_FILE_NAME;

SPOOL off
```
- The above queries gave me the following stats:
  - 4158 datasets, see /afs/cern.ch/user/v/valya/public/DBS/datasets.txt.gz
  - 60320 blocks, see /afs/cern.ch/user/v/valya/public/DBS/blocks.txt.gz
  - 1893169 files, see /afs/cern.ch/user/v/valya/public/DBS/fl-short.txt.gz

To my surprise none of the obtained LFNs have zero counts of either run or lumis, e.g. if you look at this file /afs/cern.ch/user/v/valya/public/DBS/fl-short.txt.gz you'll see something like:
```
        15        15
/store/backfill/1/express/Tier0_REPLAY_2022/ExpressPhysics/FEVT/Express-v429/000/350/966/00000/e7ca03e9-e393-4555-a5fc-8bbb4b53b948.root

        15        15
/store/backfill/1/express/Tier0_REPLAY_2022/ExpressPhysics/FEVT/Express-v429/000/350/966/00000/fea2c5a2-f164-4744-ae23-fb4d82c12125.root

         4         4
/store/backfill/1/express/Tier0_REPLAY_2022/StreamExpress/ALCAPROMPT/PromptCalibProd-Express-v429/000/350/966/00000/69abfa30-6622-4622-9811-200d3a9ec71e.root
...
```
This, weird, format came from ORACLE output, but it allowed me to scan this file using the following command `cat <file> | grep " 0"`, i.e. look for leading zero in count columns. And, I found nothing.

I also check look-up of random dataset for file,run,lumi info using dasgoclient, e.g.
```
d=/ZTo2JZTo2Jjj_EWK_QCD_LO_TuneCP5_13TeV-madgraph-pythia8/RunIISummer20UL16NanoAODv9-106X_mcRun2_asymptotic_v17-v1/NANOAODSIM
./dasgoclient -query="file,run,lumi dataset=$d"
,,,
/store/mc/RunIISummer20UL16NanoAODv9/ZTo2JZTo2Jjj_EWK_QCD_LO_TuneCP5_13TeV-madgraph-pythia8/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/196C9FB4-BCDE-CC47-AD62-E52FE0D951BC.root 1 [197,113,396]
/store/mc/RunIISummer20UL16NanoAODv9/ZTo2JZTo2Jjj_EWK_QCD_LO_TuneCP5_13TeV-madgraph-pythia8/NANOAODSIM/106X_mcRun2_asymptotic_v17-v1/70000/FCA420D1-D4F9-4940-A679-438DE7594E0D.root 1 [50,339]
```
As you can see it has run and lumis in it.

Therefore, my preliminary conclusion that within May17-Jun1st time interval we do not have problematic datasets/blocks/lfns with no run-lumis in them. It seems to me that all of our data are fine. I need more time to scan more datasets/blocks/lfns, but the initial conclusion is very encouraging.

**Comment by vkuznet** [comment]
For the record, thanks to @d-ylee we now have integration tests for concurrent data injection too, see https://github.com/dmwm/dbs2go/pull/47

**Comment by amaltaro** [comment]
@vkuznet it might be good to have someone double checking this. Can you please share 3-5 LFNs with > 500 lumis?

**Comment by vkuznet** [comment]
@amaltaro , the ORACLE format is tricky, and I adjusted query to find such patterns. Here is new query and steps I did:
```
SET LINESIZE 500
SET NEWPAGE 0
SET SPACE 0
SET PAGESIZE 0
SET ECHO OFF
SET FEEDBACK OFF
SET VERIFY OFF
SET HEADING OFF
SET TRIMSPOOL ON
SET TRIMOUT ON
SET WRAP OFF
SET TERMOUT OFF

SPOOL /tmp/fl.txt
SELECT
COUNT(FL.LUMI_SECTION_NUM) AS NLUMIS,
F.LOGICAL_FILE_NAME AS LOGICAL_FILE_NAME
FROM cms_dbs3_prod_global_owner.FILE_LUMIS FL
JOIN cms_dbs3_prod_global_owner.FILES F ON F.FILE_ID = FL.FILE_ID
WHERE 
F.creation_date>1652760000 AND
F.creation_date<1654106400 AND
F.is_file_valid=1
GROUP BY F.LOGICAL_FILE_NAME;

SPOOL off

```
- this query lists only number of lumis and LFN
- I saved results into /afs/cern.ch/user/v/valya/public/DBS/fl.txt.gz
- and, I queried this file with the following (I only select LFNs with 4 digits number of lumis):
```
zcat /afs/cern.ch/user/v/valya/workspace/fl.txt.gz | grep " [0-9][0-9][0-9][0-9]" > ~/workspace/fl-large-lumis.txt
```
You can find full dump over here: /afs/cern.ch/user/v/valya/public/DBS/fl-large-lumis.txt
It contains entries like:
```
      1014/store/mc/RunIISummer20UL17MiniAODv2/Bottomonium_oniaFilterPt9Eta1p4_muFilterPt3_TuneCP5_13TeV-pythia8-evtgen/MINIAODSIM/RAWDEBUG_106X_mc2017_realistic_v9-v1/2830000/54E7EAD1-C2E1-A54A-9539-9226E14C2076.root
      3399/store/backfill/1/data/Tier0_REPLAY_2022/MinimumBias/ALCARECO/TkAlMinBias-PromptReco-v5/000/349/840/00000/ef668d84-3f42-41f3-9347-c4ea3eed92ac.root
      1196/store/backfill/1/data/Tier0_REPLAY_2022/MinimumBias/AOD/PromptReco-v5/000/349/840/00000/005cfd2c-8ebc-45f2-90a1-068baa6ffced.root
```
and, in total we have 86 such lfns. Please check these lfns.

**Comment by vkuznet** [comment]
I did more random checks and I don't find datasets with LFNs where no lumis.

@amaltaro , Alan do I need to scan all datasets/LFNs or random sample is fine? What else do I need to do?

**Comment by amaltaro** [comment]
Sorry for the delay on getting back to this.

@vkuznet I was checking the first workflow provided by @haozturk:
https://cmsweb.cern.ch/reqmgr2/fetch?rid=pdmvserv_task_BPH-RunIISummer20UL18NanoAODv9-00186__v1_T_220514_132751_5487

and indeed it's problematic. Here is one example block with 2 LFNs:
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/files?block_name=/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/RunIISummer20UL18NanoAODv9-Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM%231e4d33df-04a4-4d59-bda1-a0b40b62bb22&detail=true

but the filelumis per block only report lumis for one of the files (note that I use dbspy here because it aggregates data):
https://cmsweb.cern.ch/dbspy/prod/global/DBSReader/filelumis?block_name=/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/RunIISummer20UL18NanoAODv9-Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM%231e4d33df-04a4-4d59-bda1-a0b40b62bb22

now checking those 2 files in different DBS APIs:
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/files?logical_file_name=/store/mc/RunIISummer20UL18NanoAODv9/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/NANOAODSIM/Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/70000/EFDC472F-93E6-EA43-BE7C-E7AA6885C5A1.root

it exists, but has no lumis:
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/filelumis?logical_file_name=/store/mc/RunIISummer20UL18NanoAODv9/promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/NANOAODSIM/Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/70000/EFDC472F-93E6-EA43-BE7C-E7AA6885C5A1.root

while the other file exists and DOES have lumis as well.

Valentin, I guess the search you need to make involves outer join or something like that. In the end, you want to look for LFNs that don't have any entry in the file_lumis table.

**Comment by vkuznet** [comment]
Alan, thanks for provided example. I look-up the dataset you are pointing out /promptDStarToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/RunIISummer20UL18NanoAODv9-Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM and checked its details. It has creation date 1652734848 which I translate into
```
>>> ts=1652734848
>>> import time
>>> time.gmtime(ts)
time.struct_time(tm_year=2022, tm_mon=5, tm_mday=16, tm_hour=21, tm_min=0, tm_sec=48, tm_wday=0, tm_yday=136, tm_isdst=0)
```
which implies that the search I made should be extended to include earliest days. As I mentioned before the creation time can be either supplied in JSON payload (and then it will be used) or it will be created by DBSWriter. As such I think this dataset was created by WMAgent on May 16 before the time I used in my search which I put as 1652760000 May 17th. Let me re-run DBS queries again use more offset in time to spot this kind of datasets.

**Comment by vkuznet** [comment]
I think I understand what we need. So far I made a JOIN query which implies that LFN FILE_ID is present in FILE_LUMIS table. What we actually need is to find LFNs whose FILE_ID is not in FILE_LUMIS table. I need to think how to craft such query.

**Comment by vkuznet** [comment]
ok, after a bit of research I came up with the following query:
```
SELECT  F.LOGICAL_FILE_NAME
FROM    cms_dbs3_prod_global_owner.FILES F
            LEFT JOIN cms_dbs3_prod_global_owner.FILE_LUMIS FL
                on F.FILE_ID = FL.FILE_ID
WHERE   FL.FILE_ID IS NULL AND
F.creation_date>1652600000 AND F.creation_date<1654106400
```
which find LFNs which do not have file_id in file_lumis table. The final list contains 6962 LFNs which  you can find here:
/afs/cern.ch/user/v/valya/public/DBS/ffl.txt

@amaltaro , Alan could you please take one of the LFNs and check it out.

**Comment by vkuznet** [comment]
And, after some work I got list of blocks via this query:
```
SELECT  B.BLOCK_NAME
FROM    cms_dbs3_prod_global_owner.FILES F
LEFT JOIN cms_dbs3_prod_global_owner.FILE_LUMIS FL ON F.FILE_ID = FL.FILE_ID
JOIN    cms_dbs3_prod_global_owner.BLOCKS B ON B.BLOCK_ID = F.BLOCK_ID
WHERE   FL.FILE_ID IS NULL AND
F.creation_date>1652600000 AND F.creation_date<1654106400
```
and datasets via processing list of blocsk as following:
```
cat /tmp/blocks-fl.txt | awk '{split($1,a,"#"); print a[1]}' | sort -n | uniq | grep ^/ > ~/public/DBS/datasets-fl.txt
```
So, the blocks and datasets can be found here:
- /afs/cern.ch/user/v/valya/public/DBS/blocks-fl.txt
- /afs/cern.ch/user/v/valya/public/DBS/datasets-fl.txt

In total, I found 1065 datasets, and 6955 blocks.

Please start your investigation and invalidation.

**Comment by vkuznet** [comment]
And for the lack of sanity check, I also checked using the following query if we do have any blocks after Jun 1st with missing FILE_LUMIS info
```
SQL> SELECT  B.BLOCK_NAME
FROM    cms_dbs3_prod_global_owner.FILES F
LEFT JOIN cms_dbs3_prod_global_owner.FILE_LUMIS FL ON F.FILE_ID = FL.FILE_ID
JOIN    cms_dbs3_prod_global_owner.BLOCKS B ON B.BLOCK_ID = F.BLOCK_ID
WHERE   FL.FILE_ID IS NULL AND
F.creation_date>1654106400  2    3    4    5    6  ;

no rows selected

```
So, the bottom line is that after Jun 1st everything is fine in DBS database.

**Comment by amaltaro** [comment]
I did a spot check on one of the files you provided and indeed it has no lumis:
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/filelumis?logical_file_name=/store/data/Run2018A/DoubleMuonLowMass/NANOAOD/UL2018_MiniAODv2_NanoAODv9_GT36-v1/60000/DE9438F9-13D7-A34C-9E7B-B9F8CFC2AE1C.root

and here is the block which it belongs to:
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/blocks?logical_file_name=/store/data/Run2018A/DoubleMuonLowMass/NANOAOD/UL2018_MiniAODv2_NanoAODv9_GT36-v1/60000/DE9438F9-13D7-A34C-9E7B-B9F8CFC2AE1C.root

I also confirm that the blockname from the output above is also present in your list of blocks in:
/afs/cern.ch/user/v/valya/public/DBS/blocks-fl.txt

And only for the record, if we ask only for unique blocks, then we are down to 2624 blocks:
```
amaltaro@lxplus716:~ $ cat  /afs/cern.ch/user/v/valya/public/DBS/blocks-fl.txt | sort | uniq | wc -l
2624
```

We need to check the next course of actions with @haozturk / ComOps, but I think we need to:
* map these 1065 output datasets to the workflow that created them
* check the status of the workflow. 
  * If it's *-archived, then we probably need to ask for the invalidation of the "empty" files in both DBS and Rucio. 
  * If it's aborted/rejected, then data should have already been invalidated in both DBS and Rucio. 
  * If it's still active in the system, P&R will decide what to do (maybe we abort/reject the workflow and its outputs; and run a brand new workflow)

This REST API: https://cmsweb.cern.ch/reqmgr2/data/request?outputdataset=/WplusToMuNu_NNPDF30_TuneEE5C_13TeV-powheg-herwigpp/RunIISummer15wmLHEGS-MCRUN2_71_V1-v1/GEN-SIM

can be used to get the workflow high level description, provided an output dataset name. Where `RequestName` contains the workflow name, and `RequestStatus` contains the current workflow status. 
Valentin, please let me know if you want me to create a script that gives us a list of workflows and their statuses.

**Comment by vkuznet** [comment]
Using Alan's suggestion I made a list of workflows affected by this issue, in total it is 316 workflows, and you can find its list over here: /afs/cern.ch/user/v/valya/public/DBS/workflows-fl.txt

**Comment by vkuznet** [comment]
and, I updated /afs/cern.ch/user/v/valya/public/DBS/blocks-fl.txt to have only unique block names, indeed, it is only 2624 of them

**Comment by vkuznet** [comment]
To make it easy for everyone I provide summary below for all links:
- list of affected workflows (316 in total): `/afs/cern.ch/user/v/valya/public/DBS/blocks-fl.txt`
- list of affected datasets (1065 in total): `/afs/cern.ch/user/v/valya/public/DBS/datasets-fl.txt`
- list of affected blocks (2624 in total): `/afs/cern.ch/user/v/valya/public/DBS/blocks-fl.txt`
- list of affected LFNs(6962 in total): `/afs/cern.ch/user/v/valya/public/DBS/ffl.txt`

**Comment by amaltaro** [comment]
Ok, looking at the list of 1065 datasets and applying this Tier0 pattern:
```
amaltaro@vocms0193:/data/amaltaro/HG2206_Val $ egrep 'Run2022|Commissioning2022' /afs/cern.ch/user/v/valya/public/DBS/datasets-fl.txt | wc -l
147
```

gives us 147 Tier0 datasets, which I made available here:
https://amaltaro.web.cern.ch/amaltaro/forWMCore/Issue_11170/t0_dsets

I guess these are a must to be recovered, right? @germanfgv @jhonatanamado 
If that's the case, we can try to recover these from the agent itself, if not yet archived.

**Comment by jhonatanamado** [comment]
Hi @amaltaro. Alan, thanks for tagging us in this issue. Tier0 operates with an `archiveddelayhours` of 7 days. That means that a lot of the workflows affected by this issue has been archived. With that in mind I believe Tier0 has two options to get recovered from this issue:

- Manually inject the old runs since this issue was spotted and invalidate only the files and blocks affected. I just stoped a tool that we use to clean old runs but I have to check from which run we can start to reprocess from scratch.

- I believe the information missing in dbs can be obtained from the files right?, If that the case proceed with a manual injection of that information into dbs.(I think this will involve a manual operation due that we need to get the information of lumis per file of the affected files)

@germanfgv What do you think?

**Comment by haozturk** [comment]
We were discussing w/ @amaltaro the plan to fix the workflows affected by this issue. Alan mentioned the following possibility where we can recover this info from the ROOT files and inject the meta-data to DBS manually:
```
1. write a script to open root files and extract run/lumi information
2. save the file + run + lumi information in a JSON output file
3. hand this JSON file to the DM team (likely DBS/Valentin team)
4. the DM team will write a script to parse the JSON file and insert the missing run/lumis - for a given file - into the database
```
From the production side, it makes sense as the other alternative is to run them from scratch and it's quite a waste of resources. We were thinking who can help us w/ fetching this info from ROOT.

**Comment by vkuznet** [comment]
@haozturk , I think I can take care of steps 3 and 4 once you'll provide file,run,lumi json output file.

**Comment by amaltaro** [comment]
I was asking Kenyi about 1., and now he volunteered himself to provide such script and run/lumi information :-D

**Comment by vkuznet** [comment]
@khurtado , yes it will be fine, i.e. create `6A7C70E9-F2DE-FF45-939E-9074677AABF4.json` with run/lumis info Let me know if you'll prepare such files.

**Comment by amaltaro** [comment]
Everyone, from a discussion we had with the T0 team this morning, what has been decided is the following:
* T0 replay data (AFAICT flagged as `REPLAY`): we do not need to recover these files, let them be as is!
* T0 data taking (AFAICT flagged as `Run2022A`): the T0  team will recover those by reprocessing that data. The current data with incomplete information in DBS will be invalidated by the T0/DM team.
* T0 commissioning data (AFAICT flagged as `Commissioning2022`): we should recover those files with the method described above (fetch data from the root file and inject it into DBS).

@drkovalskyi and @germanfgv could you please confirm and/or correct the summary above?

@khurtado please consider this when you start running your script to collect this information from the ROOT files. Dima/German mentioned that some files might no longer be accessible (already in Tape only). So we need to keep track of those and communicate it back to the T0 team.

**Comment by vkuznet** [comment]
@amaltaro I re-run the following queries:
-  to select valid lfns
```
SELECT  DISTINCT F.LOGICAL_FILE_NAME
FROM    cms_dbs3_prod_global_owner.FILES F
            LEFT JOIN cms_dbs3_prod_global_owner.FILE_LUMIS FL
                on F.FILE_ID = FL.FILE_ID
WHERE   FL.FILE_ID IS NULL
AND F.creation_date>1652600000
AND F.creation_date<1654106400
AND F.IS_FILE_VALID=1;
```
-  to select valid blocks
```
SELECT  DISTINCT B.BLOCK_NAME
FROM    cms_dbs3_prod_global_owner.FILES F
LEFT JOIN cms_dbs3_prod_global_owner.FILE_LUMIS FL ON F.FILE_ID = FL.FILE_ID
JOIN    cms_dbs3_prod_global_owner.BLOCKS B ON B.BLOCK_ID = F.BLOCK_ID
JOIN    cms_dbs3_prod_global_owner.DATASETS D ON D.DATASET_ID = B.DATASET_ID
WHERE   FL.FILE_ID IS NULL
AND F.creation_date>1652600000
AND F.creation_date<1654106400
AND F.IS_FILE_VALID=1
AND D.IS_DATASET_VALID=1;
```
- and to select datasets I run simple pipe like this:
```
cat ~/public/DBS/valid/b.txt  | awk '{split($1,a,"#"); print a[1]}' | sort | uniq > ~/public/DBS/valid/d.txt
```

The results can be found in:
- datasets `/afs/cern.ch/user/v/valya/public/DBS/valid/d.txt` (1064 in total)
- blocks `/afs/cern.ch/user/v/valya/public/DBS/valid/b.txt` (2623 in total)
- lfns `/afs/cern.ch/user/v/valya/public/DBS/valid/f.txt` (6953 in total)

Since results only differ by single digit I doubt we need to re-run all queries to get workflows.

**Comment by amaltaro** [comment]
Agreed, thanks Valentin!

**Comment by vkuznet** [comment]
In a meantime, I prepared [python script](https://gist.github.com/vkuznet/e59f6dd3c1ab7fc3128e418b75f0582b) which will generate necessary ORACLE statements to re-inject data in `FILE_LUMIS` table. I generated list of files ids, and run the script with proposed by @khurtado json file (named by LFN hash) as following:
```
./prepare.py --fin=fids.txt
```
it produced me the appropriate ORACLE statement in a file `<hash>.sql` whose content looks like this:
```
--Begin the transaction
SET TRANSACTION READ WRITE;

-- insert filelumis data
INSERT ALL
INTO cms_dbs3_prod_global_owner.FILE_LUMIS (RUN_NUM,LUMI_SECTION_NUM,FILE_ID,EVENT_COUNT) VALUES ('320809','414','888844797','0')
INTO cms_dbs3_prod_global_owner.FILE_LUMIS (RUN_NUM,LUMI_SECTION_NUM,FILE_ID,EVENT_COUNT) VALUES ('320809','415','888844797','0')
....
SELECT * FROM dual;
-- commit
COMMIT;
```
Once I'll get list of lfn-hash.json files I'll do some testing with it and we will be ready for repair procedure.

**Comment by vkuznet** [comment]
In preparation for injection procedure I asked Kate (our CERN DBA) to make a fresh dump of production DB that I can test first my injection over there. The dump will usually take some time to complete (depends on Kate's availability) and once it is ready I'll perform injection for few LFNs. Then, I'll check with dasgoclient and DBS queries to see if the affected LFNs will restore run-lumis info and report over here. Finally, when everything is ready, I'll schedule time to inject data into production DB. The scripts I'm preparing are identical in SQL statements as we have in DBS server. Therefore, the injection will be transparent to DBS operations.

**Comment by amaltaro** [comment]
Okay, I marked as injected only these 3 blocks:
```
/BuToDStarX_ToD0Pi_inclusive_SoftQCD_TuneCP5_13TeV-pythia8-evtgen/RunIISummer20UL18NanoAODv9-Custom_RDStarPU_BParking_106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM#b55338fc-46d1-4e97-b65f-1d09a9769c1f
/PMSSM_set_2_LL_2_TuneCP2_13TeV-pythia8/RunIIAutumn18MiniAOD-PUFall18Fast_GridpackScan_102X_upgrade2018_realistic_v15-v1/MINIAODSIM#ab3c8b71-80a3-461f-872c-43b2c797c3cd
/TTToSemiLeptonic_widthx0p85_TuneCP5_13TeV-powheg-pythia8/RunIISummer20UL16NanoAODv9-106X_mcRun2_asymptotic_v17-v1/NANOAODSIM#5954bf44-1a3d-4a9a-a589-6f502cf44a9f
```

because the `/TKCosmics_0T_13p6TeV...` blocks are actually pretty new - created yesterday/today - and they are failing in DBS3Upload due to a different reason. To avoid polluting too much this ticket, I decided to create a new ticket for DBS2Go project: https://github.com/dmwm/dbs2go/issues/48

**Comment by khurtado** [comment]
@amaltaro I got a few but I am submitting to condor instead, as the script has finished <1K in the list, out of >6K files. I will update here in a few hours.

**Comment by khurtado** [comment]
I have added the jsons here:

```
/afs/cern.ch/user/k/khurtado/wmcore-jsons
```

The directory has a subdirectory called `jsons` with all jsons produced. Only 2775 out of 6953 files succeeded though.
So I also have a `failedFilesMerged.txt` file with all files from `f.txt` that failed. They failed because either the file could not be found or because ROOT couldn't find the needed information in the TTrees and/or crashed.

The release used to read the files with `edmLumisInFiles.py` was `CMSSW_12_3_5`. Files were looked first locally on EOS, or via XRootD otherwise. The subdirectory: `job-logs` has all condor logs, with errors stored in `condor.err` inside each directory. I'm not sure if we can recover the files that failed, maybe a different release would work with the ones that crashed? I'm not sure, but I would need to investigate.

**Comment by vkuznet** [comment]
@khurtado , please put your stuff in your $HOME/public area otherwise we may not able to see it (due to restriction on user's file-system imposed by lxplus). Moreover, please only put json files without job/logs, etc.So far I can see jsons.tar.gz, but I can't see your wmcore area. 

@amaltaro , based on your @khurtado  input how should we proceed, i.e. he claimed that only 2775 files out of 6953 are available. I have sql's and will try to inject them in my test db which Kate prepared for me. But I need to know a final call what to do, especially with more than half LFN info is not available.

**Comment by khurtado** [comment]
@vkuznet It seems I don't have a public directory, but I just created one and made it readable using `fs sa`. It seems I didn't .
You will find the jsons in: `~/public/wmcore-jsons/wmcore/jsons`. The logfiles are in a separate subdirectory. I made all subdirs readable. Let me know if you still have problems accessing them.

**Comment by amaltaro** [comment]
@khurtado Kenyi, thanks for providing these.

Regarding the files that we failed to read and/or collect the run/lumi information, provided in your file:
failedFilesMerged.txt

I see that many of them belong to T0 replays (Tier0_REPLAY). I would suggest to remove the REPLAY files from our list of files to be recovered, as mentioned in this post:
https://github.com/dmwm/WMCore/issues/11170#issuecomment-1149908305

and try to recover what is left in your list. Can you please give this another try today?

**Comment by vkuznet** [comment]
I want to report that I successfully exercised with copy of production DBS DB, and injected several LFNs FILE_LUMIS data. Therefore, on my side everything is ready for injection.

**Comment by khurtado** [comment]
@vkuznet I'm writing a script to get the information from NANAOD (2K of the files that failed previously are of that type, so it seems no edm tool script supports it for what we need). Just to doublecheck:
We need the total number of events per lumiBlock (as opposed to a list of event numbers per lumi), correct?

E.g.:
```
lumi: 311559, nevents: 300
```
rather than:
```
lumi:31559
event: [51379560, 51379547, 51379558, 51379549...]
```

**Comment by khurtado** [comment]
Greetings,

I have produced new json files. The new file `f_new.txt`, excluding all T0 REPLAYs and Run2022A (which were reprocessed) includes 5956 files.

All files can be found in (jsons are in the `jsons` subdirectory and there is a README with a general description of the files in the directory):
```
/afs/cern.ch/user/k/khurtado/public/wmcore12
```
Summary:
In total, 5643 jsons were produced. The 313 files that failed are in `failedFilesMerged.txt`. It looks like most files that failed couldn't be found remotely.

I have uploaded all condor files and scripts I wrote in the github repository below:

https://github.com/khurtado/wmcore-jsons

[nano_lumis.py](https://github.com/khurtado/wmcore-jsons/blob/master/nano_lumis.py) : Was used for NANOAOD/NANOAODSIM files. This reads directly from the ROOT files using python ROOT using the Events TTree
[dqmio_report.py](https://github.com/khurtado/wmcore-jsons/blob/master/dqmio_report.py): Was used for DQMIO files. This reads directly from the ROOT files using python ROOT using the Indices TTree 
[aod_report.py](https://github.com/khurtado/wmcore-jsons/blob/master/aod_report.py): For everything else. This uses `edmFileUtil` and reformats the output into a json file.

Format of the jsons:

```
{"RunNumber1": {"Lumi1": TotalEvents, "Lumi2":TotalEvents}, "RunNumber2":{"Lumi1":TotalEvents}...}
```

**Comment by vkuznet** [comment]
thanks, I'm working on integration test with my own instance of DBS db, it takes a while as I walk through all steps and confirm them with our DBA.

**Comment by vkuznet** [comment]
I want to update everyone about the process which I coordinate with Kate (our DBA):
- I start the injection into my copy of production DB. This is done in chunks of 500 sql files, where each sql file represent single LFN who may have thousands of run,lumi,evt triplets.
- The data is inserted into temporary table which later will be merged into FILE_LUMIS one
- Each chunk takes about 1h to be injected into temporary table and it brings at the order of 800K run,lumi,evt triplets. Therefore, in total we will have around 10M such triplets
- Currently I'm inserting 3rd chunk and it will take me day or two to complete them
- Then, I'll need to merge my temp table into FILE_LUMIS and I expect this process may take a day or two.
- After that I can run few das queries to verify presence of run,lumis in a given LFN
- Then, we'll copy temp table to production DB and perform merge over there.

**Comment by vkuznet** [comment]
Finally, I managed to insert all except one of LFN run,lumi,evt info into FILE_LUMIS DB. The missing LFN `ED018AFE-96CB-CB42-BE14-3E4D38D67B14` with 200K run,lumi rows is not yet injected as I'll need to handle it separately. As such, now the run-lumi information is available for LFNs in question and you can easily check it as following:
```
lfn=/store/data/Run2018D/SingleMuon/NANOAOD/UL2018_MiniAODv1_NanoAODv2_GT36-v1/60000/ED029895-5B6C-D345-870F-1784F4105B78.root

# then query DBS
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/filelumis?logical_file_name=$lfn

# and results are
[
{"event_count":3085,"logical_file_name":"/store/data/Run2018D/SingleMuon/NANOAOD/UL2018_MiniAODv1_NanoAODv2_GT36-v1/60000/ED029895-5B6C-D345-870F-1784F4105B78.root","lumi_section_num":60,"run_num":321975}
...
,{"event_count":4613,"logical_file_name":"/store/data/Run2018D/SingleMuon/NANOAOD/UL2018_MiniAODv1_NanoAODv2_GT36-v1/60000/ED029895-5B6C-D345-870F-1784F4105B78.root","lumi_section_num":19,"run_num":325169}
]
```

**Comment by khurtado** [comment]
That's great news!

**Comment by amaltaro** [comment]
Kenyi, Valentin, thank you very much for sorting this out.

> In total, 5643 jsons were produced. The 313 files that failed are in failedFilesMerged.txt. It looks like most files that failed couldn't be found remotely.

I had a quick look at the list of files that we didn't succeed recovering, in:
/afs/cern.ch/user/k/khurtado/public/wmcore12/failedFilesMerged.txt

and all of them are actually `Commissioning2022`.

I did an spot check on:
/NoBPTX/Commissioning2022-v1/RAW

and it seems to only be available on Tape endpoints. I am also not sure whether T0 Disk is accessible through xrootd(?)

@germanfgv @jhonatanamado I guess it's too late to try to rerun these, right? I believe this information is no longer available in the agent database, but we should check whether these files can still be found in the agent, and in that case we might be able to get a list of run/lumis/evts for those missing files.

**Comment by jhonatanamado** [comment]
Hi @amaltaro I would suggest another idea, Due that these files are only on tape and indeed is too late to try to reprocess them, what about to ask DM to transfer the files affected to a Disk site and proceed to get the information and populate dbs with the missing information. If Im not wrong, is possible to create temporary rule over each file instead of the whole block to which the file belongs.

**Comment by amaltaro** [comment]
If we can transfer only those files, then yes, we could try to recover it that way. @khurtado what do you think?

**Comment by khurtado** [comment]
I'm on board with it. Let's try!

**Comment by flgomezc** [comment]
Hello @khurtado @amaltaro,
Following @jhonatanamado, I have created rules to transfer the files to T2_CH_CERN.
The files will be there 15 days.
Cheers!

**Comment by jhonatanamado** [comment]
Thanks @flgomezc I checked with rucio and the state of the rules for the single files are ok.  @khurtado. You have green light to run again your script.

**Comment by khurtado** [comment]
@jhonatanamado Awesome, thanks!

@vkuznet @amaltaro : All missing files have been recovered. Jsons are in:

```
/afs/cern.ch/user/k/khurtado/public/jsons_missing313
```
I didn't put the logfiles because all of them succeeded anyway.

**Comment by vkuznet** [comment]
@amaltaro , @khurtado I successfully injected the rest of 313 files. Now everything should be covered. Please let me know if anything is missing in this ticket.

**Comment by khurtado** [comment]
@vkuznet Thanks!. That was a 100% recovery then. Good job everyone! @amaltaro: I vote for closing this issue then.

**Comment by amaltaro** [comment]
Great! Thanks to everyone that contributed to solving this problem!
